test/integration/web_console_access_test.go

```diff
@@ -0,0 +1,96 @@
+// +build integration,!no-etcd
+
+package integration
+
+import (
+	"crypto/tls"
+	"encoding/json"
+	"io/ioutil"
+	"net/http"
+	"testing"
+
+	configapi "github.com/openshift/origin/pkg/cmd/server/api"
+	testutil "github.com/openshift/origin/test/util"
+)
+
+func tryAccessURL(t *testing.T, url string, expectedStatus int, expectedRedirectLocation string) *http.Response {
+	transport := &http.Transport{
+		TLSClientConfig: &tls.Config{
+			InsecureSkipVerify: true,
+		},
+	}
+
+	req, err := http.NewRequest("GET", url, nil)
+	req.Header.Set("Accept", "text/html")
+	resp, err := transport.RoundTrip(req)
+	if err != nil {
+		t.Errorf("Unexpected error while accessing %s: %v", url, err)
+		return nil
+	}
+	if resp.StatusCode != expectedStatus {
+		t.Errorf("Expected status %d for %s, got %d", expectedStatus, url, resp.StatusCode)
+	} else {
+		if expectedRedirectLocation != "" {
```

Comment:
actually, the check shouldn't even be conditional... if we expect `""`, and get `"foo"`, that's an error